ipn/ipnlocal/local_test.go

```diff
@@ -3247,104 +3200,57 @@ func TestSuggestExitNode(t *testing.T) {
+		wantID     tailcfg.StableNodeID
 	}{
 		{
-			name: ">1 candidates",
+			name: "different priorities",
 			candidates: []tailcfg.NodeView{
-				(&tailcfg.Node{
-					ID:       2,
-					StableID: "2",
-					AllowedIPs: []netip.Prefix{
-						netip.MustParsePrefix("0.0.0.0/0"), netip.MustParsePrefix("::/0"),
-					},
-					Hostinfo: (&tailcfg.Hostinfo{
-						Location: &tailcfg.Location{
-							Priority: 20,
-						},
-					}).View(),
-				}).View(),
-				(&tailcfg.Node{
-					ID:       3,
-					StableID: "3",
-					AllowedIPs: []netip.Prefix{
-						netip.MustParsePrefix("0.0.0.0/0"), netip.MustParsePrefix("::/0"),
-					},
-					Hostinfo: (&tailcfg.Hostinfo{
-						Location: &tailcfg.Location{
-							Priority: 10,
-						},
-					}).View(),
-				}).View(),
+				makePeer(2, withExitRoutes(), withLocation(location20.View())),
+				makePeer(3, withExitRoutes(), withLocation(location10.View())),
 			},
-			wantValue: (&tailcfg.Node{
-				ID:       2,
-				StableID: "2",
-				AllowedIPs: []netip.Prefix{
-					netip.MustParsePrefix("0.0.0.0/0"), netip.MustParsePrefix("::/0"),
-				},
-				Hostinfo: (&tailcfg.Hostinfo{
-					Location: &tailcfg.Location{
-						Priority: 20,
-					},
-				}).View(),
-			}).View(),
-			wantValid: true,
+			wantID: "stable2",
+		},
+		{
+			name: "same priorities",
+			candidates: []tailcfg.NodeView{
+				makePeer(2, withExitRoutes(), withLocation(location10.View())),
+				makePeer(3, withExitRoutes(), withLocation(location10.View())),
+			},
+			wantID: "stable2",
 		},
 		{
 			name:       "<1 candidates",
 			candidates: []tailcfg.NodeView{},
-			wantValid:  false,
+			wantID:     "<invalid>",
 		},
 		{
 			name: "1 candidate",
 			candidates: []tailcfg.NodeView{
-				(&tailcfg.Node{
-					ID:       2,
-					StableID: "2",
-					AllowedIPs: []netip.Prefix{
-						netip.MustParsePrefix("0.0.0.0/0"), netip.MustParsePrefix("::/0"),
-					},
-					Hostinfo: (&tailcfg.Hostinfo{
-						Location: &tailcfg.Location{
-							Priority: 20,
-						},
-					}).View(),
-				}).View(),
+				makePeer(2, withExitRoutes(), withLocation(location20.View())),
 			},
-			wantValue: (&tailcfg.Node{
-				ID:       2,
-				StableID: "2",
-				AllowedIPs: []netip.Prefix{
-					netip.MustParsePrefix("0.0.0.0/0"), netip.MustParsePrefix("::/0"),
-				},
-				Hostinfo: (&tailcfg.Hostinfo{
-					Location: &tailcfg.Location{
-						Priority: 20,
-					},
-				}).View(),
-			}).View(),
-			wantValid: true,
+			wantID: "stable2",
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got := pickWeighted(tt.candidates)
-			if !reflect.DeepEqual(got, tt.wantValue) {
-				t.Errorf("got value %v want %v", got, tt.wantValue)
-				if tt.wantValid != got.Valid() {
-					t.Errorf("got invalid candidate expected valid")
-				}
-				if tt.wantValid {
-					if !reflect.DeepEqual(got, tt.wantValue) {
-						t.Errorf("got value %v want %v", got, tt.wantValue)
-					}
-				}
+			gotID := tailcfg.StableNodeID("<invalid>")
```

Comment:
This is kind of awkward but I have a follow on PR that's going to make `pickWeighted` return a slice in case there's a tie, and this approach makes more sense then.